pkg/sbom/spdx/marshal.go

```diff
@@ -113,9 +112,9 @@ func NewMarshaler(version string, opts ...marshalOption) *Marshaler {
 	return m
 }
 
-func (m *Marshaler) Marshal(r types.Report) (*spdx.Document2_2, error) {
-	var relationShips []*spdx.Relationship2_2
-	packages := make(map[spdx.ElementID]*spdx.Package2_2)
+func (m *Marshaler) Marshal(r types.Report) (*spdx.Document, error) {
+	var relationShips []*spdx.Relationship
+	packages := make(map[spdx.ElementID]*spdx.Package)
```

Comment:
I think we can change it to `[]*spdx.Package`.

```suggestion
	packages := []*spdx.Package
```

Reply:
Could you share an example where you saw the duplications?

Reply:
Sure, I've got it from the unit test `TestMarshaler_Marshal - happy path for container scan`, `actionpack` define in `Target: "app/subproject/Gemfile.lock",` and in `Target: "app/Gemfile.lock",`
```
 marshal_test.go:770: Marshal mismatch (-want +got):
          &v2_3.Document{
          	... // 6 identical fields
          	DocumentComment: "",
          	CreationInfo:    &{Creators: {{Creator: "aquasecurity", CreatorType: "Organization"}, {Creator: "trivy-0.38.1", CreatorType: "Tool"}}, Created: "2021-08-25T12:20:30Z"},
          	Packages: []*v2_3.Package{
          		... // 2 identical elements
          		&{PackageName: "rails:latest", PackageSPDXIdentifier: "ContainerImage-9396d894cd0cb6cb", PackageDownloadLocation: "NONE", PackageExternalReferences: {&{Category: "PACKAGE-MANAGER", RefType: "purl", Locator: "pkg:oci/rails@sha256:a27fd8080b517143cbbbab9dfb7c8571c40d67d534b"...}}, ...},
          		&{PackageName: "centos", PackageSPDXIdentifier: "OperatingSystem-197f9a00ebcb51f0", PackageVersion: "8.3.2011", PackageDownloadLocation: "NONE", ...},
        + 		&{
        + 			PackageName:             "actionpack",
        + 			PackageSPDXIdentifier:   "Package-826226d056ff30c0",
        + 			PackageVersion:          "7.0.1",
        + 			PackageDownloadLocation: "NONE",
        + 			PackageLicenseConcluded: "NONE",
        + 			PackageLicenseDeclared:  "NONE",
        + 			PackageExternalReferences: []*v2_3.PackageExternalReference{
        + 				&{
        + 					Category: "PACKAGE-MANAGER",
        + 					RefType:  "purl",
        + 					Locator:  "pkg:gem/actionpack@7.0.1",
        + 				},
        + 			},
        + 		},
          		&{PackageName: "actionpack", PackageSPDXIdentifier: "Package-826226d056ff30c0", PackageVersion: "7.0.1", PackageDownloadLocation: "NONE", ...},
          		&{PackageName: "actioncontroller", PackageSPDXIdentifier: "Package-eb0263038c3b445b", PackageVersion: "7.0.1", PackageDownloadLocation: "NONE", ...},
          		&{PackageName: "binutils", PackageSPDXIdentifier: "Package-fd0dc3cf913d5bc3", PackageVersion: "2.30-93.el8", PackageDownloadLocation: "NONE", ...},
          	},
```

Reply:
Thanks. That is what I expected, and I wanted to confirm it. I want to change it to have both files, but it can be done in another PR. We can leave it as is.